Classes/Fusion/Helper/WorkspaceHelper.php

```diff
@@ -51,15 +52,48 @@ public function getPersonalWorkspace()
     {
         $personalWorkspace = $this->userService->getPersonalWorkspace();
         $baseWorkspace = $personalWorkspace->getBaseWorkspace();
+        $readOnly = !$this->domainUserService->currentUserCanPublishToWorkspace($baseWorkspace);
+
+        if ($readOnly === true && $this->workspaceService->shouldHideReadOnlyWorkspaces()) {
+            // Skip read only workspaces
+            $baseWorkspace = $this->getFirstWriteableTargetWorkspace();
+            $this->workspaceService->setBaseWorkspace($baseWorkspace);
+            $readOnly = !$this->domainUserService->currentUserCanPublishToWorkspace($baseWorkspace);
```

Comment:
```suggestion
        $readOnly = !$this->domainUserService->currentUserCanPublishToWorkspace($baseWorkspace);

        if ($readOnly === true && $this->workspaceService->shouldHideReadOnlyWorkspaces()) {
            // Skip read only workspaces
            $baseWorkspace = $this->getFirstWriteableTargetWorkspace();
            $this->workspaceService->setBaseWorkspace($baseWorkspace);
```

Line 61 seems to be a duplicate line. $readOnly is overridden with the same? value again.

Reply:
In the most cases it will be the same.
But when there is no writable workspace, we use the configured targetWorkspace as fallback and this one can have another value. For a restricted editor, the live workspace is not writable for instance.

Reply:
Thanks for the explanation, i missed $baseWorkspace  being changed ^^